tests/components/backup/test_backup.py

```diff
@@ -119,11 +154,13 @@ async def test_delete_backup(
 
     with (
         patch("pathlib.Path.exists", return_value=backup_exists),
-        patch("pathlib.Path.unlink") as unlink,
+        patch("os.unlink") as unlink,
```

Comment:
Why is this changed?

Reply:
I changed it to be able to check the filename in the `os.unlink`call from `Path.unlink`, since the mock calls don't capture the self-parameter when just doing `patch("os.unlink") as unlink`.
I can change it to `patch("pathlib.Path.unlink", autospec=True) as unlink,`, maybe that's clearer.

Reply:
Yeah, the latter is a bit clearer.